src/NLog/Targets/FileTarget.cs

```diff
@@ -615,7 +586,7 @@ protected override void CloseTarget()
         {
             base.CloseTarget();
 
-            foreach (string fileName in new List<string>(this.initializedFiles.Keys))
+            foreach (string fileName in initializedFiles.GetItems())
```

Comment:
This can be problematic? ("Collection was modified; enumeration operation may not execute")?